cmd/filter/osdiskfilter.go

```diff
@@ -0,0 +1,103 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/node-disk-manager/cmd/controller"
+	"github.com/openebs/node-disk-manager/pkg/util"
+)
+
+const (
+	osDiskFilterName  = "os disk filter" // osDiskFilter name
+	osDiskFilterState = defaultEnabled   // osDiskFilter state
+)
+
+var (
+	defaultMountFilePath = "/proc/self/mounts"
+	mountPoints          = []string{"/", "/etc/hosts"}
+	hostMountFilePath    = "/host/mounts" // hostMountFilePath is the file path mounted inside container
+)
+
+// osDiskFilterRegister contains registration process of osDiskFilter
+var osDiskFilterRegister = func() {
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		return
+	}
+	var fi controller.FilterInterface = newOsDiskFilter(ctrl)
+	newPrgisterFilter := &registerFilter{
+		name:       osDiskFilterName,
+		state:      osDiskFilterState,
+		fi:         fi,
+		controller: ctrl,
+	}
+	newPrgisterFilter.register()
+}
+
+// osDiskFilter controller and uuid of os disk
+type osDiskFilter struct {
+	controller *controller.Controller
+	devPath    string
+}
+
+// newOsDiskFilter returns new pointer osDiskFilter
+func newOsDiskFilter(ctrl *controller.Controller) *osDiskFilter {
+	return &osDiskFilter{
+		controller: ctrl,
+	}
+}
+
+// Start set os disk devPath in osDiskFilter pointer
+func (odf *osDiskFilter) Start() {
+	// process1 check for mentioned mountpoint in host's /proc/1/mounts file
+	// host's /proc/1/mounts file mounted inside container it checks for
+	// every mentioned mountpoints if it is able to get parent disk's uuid
+	// it adds it in Controller struct and make isOsDiskFilterSet true
+	for _, mountPoint := range mountPoints {
+		mountPointUtil := util.NewMountUtil(hostMountFilePath, mountPoint)
+		if devPath, err := mountPointUtil.GetDiskPath(); err == nil {
+			odf.devPath = devPath
+			return
+		}
+	}
+	// process2 check for mountpoints in /proc/self/mounts file
+	// if it is able to get disk's uuid of it adds it in Controller struct
+	// and make isOsDiskFilterSet true
+	for _, mountPoint := range mountPoints {
+		mountPointUtil := util.NewMountUtil(defaultMountFilePath, mountPoint)
+		if devPath, err := mountPointUtil.GetDiskPath(); err == nil {
+			odf.devPath = devPath
+			return
+		}
+	}
+	glog.Error("unable to apply os disk filter")
+}
+
+// Include checks if disk uuid of passing DiskInfo struct is same as
+// uuid mention in osDiskFilter struct if it matches it returns true
+// else it returns false
+func (odf *osDiskFilter) Include(d *controller.DiskInfo) bool {
+	return odf.devPath == d.Path
+}
+
+// Include checks if disk uuid of passing DiskInfo struct is same as
```

Comment:
Fix these comments as they apply to OSdisk

Reply:
done